File: ios/Src/RNIPopoverView/RNIPopoverViewManager.m
//
//  RCTPopoverViewManager.m
//  IosPopoverExample
//
//  Created by Dominic Go on 11/27/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import <React/RCTViewManager.h>
 

@interface RCT_EXTERN_MODULE(RNIPopoverViewManager, RCTViewManager)

// ---------------------------
// MARK: Export Props - Events
// ---------------------------

RCT_EXPORT_VIEW_PROPERTY(onPopoverWillShow, RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPopoverWillHide, RCTBubblingEventBlock);

RCT_EXPORT_VIEW_PROPERTY(onPopoverDidShow, RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPopoverDidHide, RCTBubblingEventBlock);

RCT_EXPORT_VIEW_PROPERTY(onPopoverWillHideViaTap, RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPopoverDidHideViaTap , RCTBubblingEventBlock);

RCT_EXPORT_VIEW_PROPERTY(onPopoverDidAttemptToDismiss, RCTBubblingEventBlock);

// ---------------------------
// MARK: Export Props - Values
// ---------------------------

RCT_EXPORT_VIEW_PROPERTY(popoverSize, NSDictionary);
RCT_EXPORT_VIEW_PROPERTY(permittedArrowDirections, NSArray);
RCT_EXPORT_VIEW_PROPERTY(popoverBackgroundColor, NSNumber);
RCT_EXPORT_VIEW_PROPERTY(internalCleanupMode, NSString);

RCT_EXPORT_VIEW_PROPERTY(popoverShouldDismiss, BOOL);
RCT_EXPORT_VIEW_PROPERTY(popoverCanOverlapSourceViewRect, BOOL);

@end

File: ios/Src/ReactNative/RNIPopoverView/RNIPopoverViewManager.swift
//
//  RCTPopoverViewManager.swift
//  IosPopoverExample
//
//  Created by Dominic Go on 11/27/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

@objc(RNIPopoverViewManager)
class RNIPopoverViewManager: RCTViewManager {
  
  override static func requiresMainQueueSetup() -> Bool {
    return true;
  };
  
  override func view() -> UIView! {
    return RNIPopoverView(bridge: self.bridge);
  };
  
  @objc func setVisibility(_ node: NSNumber, visibility: Bool){
    DispatchQueue.main.async {
      guard let view = self.bridge.uiManager.view(forReactTag: node),
            let popoverView = view as? RNIPopoverView
      else { return };
      
      popoverView.setVisibility(visibility);
    };
  };
};
